### ISSUE TITLE
Hibernate doesn't use autoconfigured HazelcastInstance for 2nd Level Cache

### ISSUE THREAD

**Issue by vpavic** [spring-projects/spring-boot#4158]
I'm trying to use ```HazelcastInstance``` autoconfigured by Spring Boot as Hibernate 2nd Level Cache by defining the following in ```hibernate.properties```:

```properties
hibernate.cache.region.factory_class=com.hazelcast.hibernate.HazelcastCacheRegionFactory
hibernate.cache.hazelcast.instance_name=defaultInstance
```

However, Spring Boot will initialize ```EntityManagerFactory``` before ```HazelcastInstance```, resulting with configuration being loaded from ```hazelcast-default.xml``` instead of provided ```Config``` ```@Bean```.

Minimal project that demonstrates the issue is available here:
https://github.com/vpavic/hibernate-hazelcast-config

I have tried to add```HibernateJpaAutoConfiguration``` as ```@AutoConfigureBefore``` on ```HazelcastAutoConfiguration```, and ```HazelcastAutoConfiguration``` as ```@AutoConfigureAfter``` on ```HibernateJpaAutoConfiguration```, but to no avail.

It would be nice to be able use such configuration without manually configuring ```EntityManagerFactory```.

**Comment by snicoll** [comment]
I didn't thought of that one, thanks for the sample project

**Comment by vpavic** [comment]
@snicoll, thanks for the prompt fix!

Having tested the latest 1.3.0.BUILD-SNAPSHOT in my projects I can confirm that the issue is resolved.

**Comment by snicoll** [comment]
:+1: thanks for testing!

**Comment by snicoll** [comment]
Unfortunately, that post processor is added unconditionally so if you happen to have hazelcast on your classpath with no intention of using it, this blows up because we're trying to depend on a bean that does not exist

**Comment by snicoll** [comment]
@vpavic I had to rework the initial commit. If you have a chance to test master once the snapshot has been deployed that would be great.

**Comment by vpavic** [comment]
Thanks @snicoll, I'll test against the snapshot this evening and get back to you.

**Comment by vpavic** [comment]
I can confirm the issue is fixed with the latest 1.3.0.BUILD-SNAPSHOT.

The configured ```HazelcastInstance``` is used both by Hibernate 2nd Level Cache and ```HazelcastCacheManager```.

**Comment by snicoll** [comment]
Thank you!
